File: Java/ContainerWater.java
// prove by contradiction, move the smaller cursor will not eliminate the right answer
public int maxArea(int[] height) {
    if (height == null)
        return 0;
    int result = 0;
    int left = 0, right = height.length - 1;
    while (left < right) {
        result = Math.max(Math.min(height[left], height[right])*(right - left), result);
        if (height[left] > height[right])
            right--;
        else
            left++;
    }
    return result;
}

File: Python/ContainerWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        if len(height) < 2:
            return 0
        result = 0
        i, j = 0, len(height) - 1
        while i < j:
            if height[i] < height[j]:
                result = max(result, height[i]*(j-i))
                i += 1
            else:
                result = max(result, height[j]*(j-i))
                j -= 1                
        return result
